2009-02-13  Michael Tautschnig  <mt@debian.org>

	* lib/get-config-dir-*: task_error requires the exit code of the preceeding
		command as the second argument
	* lib/subroutines-linux: The exit code of a command earlier in a pipe must be
		read using $PIPESTATUS, not $? (closes: #444210)
Index: trunk/lib/get-config-dir-cvs
===================================================================
--- trunk.orig/lib/get-config-dir-cvs
+++ trunk/lib/get-config-dir-cvs	
@@ -36,11 +36,11 @@
    echo "Updating CVS in $FAI"
    cd $FAI
    cvs -q up -P $tag -d -C > $LOGDIR/getconf.log
-   task_error 701
+   task_error 701 $?
 else 
    echo "Checking out CVS"
    cd $(dirname "$FAI")
    cvs -q co -P -d $(basename "$FAI") \
      $tag $module > $LOGDIR/getconf.log
-   task_error 702
+   task_error 702 $?
 fi
Index: trunk/lib/get-config-dir-git
===================================================================
--- trunk.orig/lib/get-config-dir-git
+++ trunk/lib/get-config-dir-git	
@@ -32,9 +32,9 @@
    echo "Updating git copy in $FAI"
    cd $FAI
    git pull
-   task_error 701
+   task_error 701 $?
 else 
    echo "Checking out from git"
    git clone $giturl $FAI 
-   task_error 702
+   task_error 702 $?
 fi
Index: trunk/lib/get-config-dir-nfs
===================================================================
--- trunk.orig/lib/get-config-dir-nfs
+++ trunk/lib/get-config-dir-nfs	
@@ -12,4 +12,4 @@
 
 mount -n $romountopt $server:$nfspath $FAI &&
   echo "Configuration space $server:$nfspath mounted to $FAI"
-task_error 701
+task_error 701 $?
Index: trunk/lib/get-config-dir-svn
===================================================================
--- trunk.orig/lib/get-config-dir-svn
+++ trunk/lib/get-config-dir-svn	
@@ -50,9 +50,9 @@
    echo "Updating SVN in $FAI"
    cd $FAI
    svn up $user | grep -v 'Updated to revision' > $LOGDIR/getconf.log
-   task_error 701
+   task_error 701 $?
 else 
    echo "Checking out SVN"
    svn co $user $svnurl $FAI | grep -v 'Checked out revision' > $LOGDIR/getconf.log
-   task_error 702
+   task_error 702 $?
 fi
Index: trunk/lib/subroutines-linux
===================================================================
--- trunk.orig/lib/subroutines-linux
+++ trunk/lib/subroutines-linux	
@@ -250,10 +250,10 @@
     echo "Installing software may take a while"
     if [ "$debug" ]; then
 	install_packages | tee -a $LOGDIR/software.log
-	task_error 471 $?
+	task_error 471 ${PIPESTATUS[0]}
     elif [ "$verbose" ]; then
 	install_packages </dev/null 2>&1 | tee -a $LOGDIR/software.log
-	task_error 471 $?
+	task_error 471 ${PIPESTATUS[0]}
     else
 	install_packages </dev/null >> $LOGDIR/software.log 2>&1
 	task_error 471 $?
